packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -591,6 +591,79 @@ The eventual response back from the pre-authz service should resolve to somethin
 }
 ```
 
+# Authentication Refresh Service
+Since synchronization of a user's session is important to provide a seamless user experience when using an app and transacting with the Flow Blockchain, a way to confirm, extend, and refresh a user session can be provided by the wallet.
+
+Authentication Refresh Services should include a `type: "authn-refresh"`, `endpoint`, and supported `method` (`HTTP/POST`, `IFRAME/RPC`, `POP/RPC`, etc).
+
+FCL will use the `endpoint` and service `method` provided to request updated authentication data.
+The `authn-refresh` service should refresh the user's session if neccessary and return updated authentication configuration and user session data.
+
+The service is expected to return a `PollingResponse` with a new `AuthnResponse` as data. If user input is required, a `PENDING` `PollingResponse` can be returned with a `local` view for approval/re-submission of user details.
```

Comment:
If the `authn-refresh` service is called in the background, and a local view is rendered, it may interrupt the user flow in the app. Should we highlight that point that requiring a local should be avoided if possible?

Reply:
I haven't tested this, yet. Was under the impression that this should work with any service method and flow. 
Call to Authn Refresh happens before authz, etc. So if a local is used and user needs to approve, flow should continue and return back to authn-refresh service. @orodio Any thoughts on this?

Reply:
Update @JeffreyDoyle @orodio is I've tested using our [mock http api](https://github.com/onflow/fcl-http-post) which returns a `local` view and `updates` endpoint for `authn`. This does not interrupt `authn-refresh` flow when including user interaction via iFrame. Updated user is returned and rebuilt, `authz` service is then executed with latest `currentUser`